src/apscheduler/_schedulers/async_.py

```diff
@@ -529,6 +532,64 @@ async def remove_schedule(self, id: str) -> None:
         self._check_initialized()
         await self.data_store.remove_schedules({id})
 
+    async def pause_schedule(self, schedule_id: str) -> None:
+        """Pause the specified schedule."""
+        self._check_initialized()
+        await self.data_store.add_schedule(
+            schedule=attrs.evolve(await self.get_schedule(schedule_id), paused=True),
+            conflict_policy=ConflictPolicy.replace,
+        )
+
+    def _get_unpaused_next_fire_time(
+        self,
+        schedule: Schedule,
+        resume_from: datetime | Literal["now"] | None,
+    ) -> datetime | None:
+        if resume_from is None:
+            return schedule.next_fire_time
+        if resume_from == "now":
+            resume_from = datetime.now(tz=timezone.utc)
+        if (
+            schedule.next_fire_time is not None
+            and schedule.next_fire_time >= resume_from
+        ):
+            return schedule.next_fire_time
+        try:
+            while (next_fire_time := schedule.trigger.next()) < resume_from:
+                pass  # Advance `next_fire_time` until its at or past `resume_from`
+        except TypeError:  # The trigger is exhausted
+            return None
+        return next_fire_time
```

Comment:
Add a blank line after each control block to visually distinguish it from others.

Reply:
Also, I don't like catching `TypeError` here, as it has the potential to mask actual bugs. I'd prefer an explicit check against a `None` return value, even if it adds a line or two of code.